dsn_test.go

```diff
@@ -1639,3 +1639,19 @@ func TestExtractAccountName(t *testing.T) {
 		})
 	}
 }
+
+func TestDecodeDSNParam(t *testing.T) {
```

Comment:
If the purpose of this function is to unit test a single method, maybe we can name it `TestUrlDecodeIfNeeded`?

Reply:
that is indeed the purpose, renaming